scalingo/data_scalingo_invoices.go

```diff
@@ -0,0 +1,213 @@
+										Computed: true,
+									},
+								},
+							},
+						},
+						"detailed_items": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"label": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"price": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"app": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+const PageSize = 50
+
+func isInTimeRange(before time.Time, after time.Time, value time.Time) bool {
+	return value.After(before) && value.Before(after)
+}
+
+func structToMap(v interface{}) (map[string]interface{}, error) {
+	var rawStruct map[string]interface{}
+	jsonStruct, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(jsonStruct, &rawStruct)
+	if err != nil {
+		return nil, err
+	}
+	return rawStruct, err
+}
+
+func dataSourceScInvoiceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var beforeTime time.Time
+	var afterTime time.Time
+	var err error
+
+	client, _ := meta.(*scalingo.Client)
+
+	// handling config and setting default values if empty
+	beforeTimeStr, _ := d.Get("before").(string)
+	if beforeTimeStr == "" {
+		beforeTime = time.Unix(0, 0)
+	} else {
+		beforeTime, err = time.Parse(scalingo.BillingMonthDateFormat, beforeTimeStr)
+		if err != nil {
+			return diag.Errorf("fail to parse before: %v", err)
+		}
+	}
+
+	afterTimeStr, _ := d.Get("after").(string)
+	if afterTimeStr == "" {
+		afterTime = time.Now().Add(time.Hour * 24)
+	} else {
+		afterTime, err = time.Parse(scalingo.BillingMonthDateFormat, afterTimeStr)
+		if err != nil {
+			return diag.Errorf("fail to parse after: %v", err)
+		}
+		afterTime = afterTime.Add(time.Hour * 24)
+	}
+
+	// fetch all invoices in a slice
+	maxPage := 1
+	currentPage := 1
+	var invoices []*scalingo.Invoice
+	for currentPage <= maxPage {
+		pageInvoices, pagination, err := client.InvoicesList(ctx, scalingo.PaginationOpts{
+			Page:    currentPage,
+			PerPage: PageSize,
+		})
+		if err != nil {
+			return diag.Errorf("fail to list invoices: %v", err)
+		}
+		if currentPage == 1 {
+			maxPage = pagination.TotalPages
+			invoices = make([]*scalingo.Invoice, 0, pagination.TotalCount)
+		}
+		invoices = append(invoices, pageInvoices...)
+		currentPage++
+	}
+
+	// filtering invoices with the current config
+	filteredInvoices := keepIf(invoices, func(invoice *scalingo.Invoice) bool {
+		return isInTimeRange(beforeTime, afterTime, time.Time(invoice.BillingMonth))
+	})
+
+	// mapping invoices list to raw json struct before saving in the state to keep json fields
+	invoicesState, err := structToMap(map[string]interface{}{
```

Comment:
This call doesn't work without an update in go-scalingo allowing to marshal BillingMonth as a string . What is the best option: 
- handling it in go-scalingo by adding the Marshal [interface](https://github.com/Scalingo/go-scalingo/blob/077494ebff146a6917d6edc9a39b5f2bfc25e9d9/invoices.go#L92) 
- writing some code to marshal the struct by hand

Reply:
I think it would be coherent to add the Marshal method because we already have the Unmarshal for the same field

Reply:
Decision was taken to handle this locally